source/MRMesh/MRDeferred.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <utility>
+
+// TODO: move to separate header
+#define MR_CONCAT_IMPL( A, B ) A##B
+#define MR_CONCAT( A, B ) MR_CONCAT_IMPL( A, B )
+
+namespace MR
+{
+
+/// helper class to call a function at its scope's exit (RAII-style)
+template <typename F>
+class DeferredFunction
+{
+    F f_;
+
+public:
+    DeferredFunction( F&& f )
+        : f_( std::forward<F>( f ) )
```

Comment:
Here `F&& f` is rvalue-reference and not universal reference, so you can replace `std::forward<F>` with `std::move`

Reply:
> `F&& f` is rvalue-reference

This is correct for `MR_DEFER_INLINE`, but not for `MR_DEFER`; you might pass any functor to it, including const objects and function pointers.

Reply:
I see, `F` can be already a const reference.